packages/client/src/declarations/index.ts

```diff
@@ -264,11 +264,14 @@ type RelationForCertificateCorrection =
   | 'CHILD'
 
 export type ICertificate = {
-  collector?: Partial<{ type: Relation | string }>
+  collector?: Partial<{
+    type: Relation | string
+    certificateTemplateId?: string
```

Comment:
Yes, as shown in the screenshot, in the collector form, there is no longer only one field (type), but rather two fields (type and certificateTemplateId).
In the collector form, this form's value resides inside sectionData which is an `ICertificate` type. To access the `type` (collector type) value as well as `certificateTemplateId` (selected certificate template), I needed to add this optional property here.

Reply:
Is there another way of doing it? I'm only considering the data model here where we are now saying that a Collector has `certificateTemplateId`. Doesn't that feel a little off?

Reply:
Umm, I would like to think it in this way, "Which certificate template did the collector use to print this certificate?"

Reply:
Ok, in that case, let's keep it under collector and make it mandatory and remove the other `certificateTemplateId?: string` from line 276

Reply:
Then again, this interface `ICertificate` defines the `IPrintableDeclaration` interface's `certificates` property on line no 299. In the `IssuCollectorForm` component, when we try to modifyDeclaration, the declaration type is `IPrintableDeclaration`, and from the backend, the downloaded declaration data contains `certificateTemplateId` in `registration.certificates` array.
So we kind of need `certificateTemplateId` outside the collector's scope, since `ICertificate` interface is used in various scenarios.

Reply:
Either way is fine as long as we only have it in one place 👍